src/FMU_Wrapper.f90

```diff
@@ -0,0 +1,753 @@
+    GENERIC :: getNamedVariable => getNamedReal_FMU2_Slave, &
+        getNamedInteger_FMU2_Slave, getNamedBoolean_FMU2_Slave
+    GENERIC :: setNamedVariable => setNamedReal_FMU2_Slave, &
+        setNamedInteger_FMU2_Slave, setNamedBoolean_FMU2_Slave
+ENDTYPE FMU2_Slave
+
+!> Exception Handler for use in MatrixTypes
+TYPE(ExceptionHandlerType),SAVE :: eFMU_Wrapper
+
+!> Name of module
+CHARACTER(LEN=*),PARAMETER :: modName='FMU_Wrapper'
+
+!
+!===============================================================================
+CONTAINS
+!
+!-------------------------------------------------------------------------------
+!> @brief Initializes the FMU Slave
+!>
+!> @param self the FMU2_Slave to act on
+!> @param id Unique id for this FMU
+!> @param pList A parameter list with input options
+!>
+SUBROUTINE init_FMU2_Slave(self,id,pList)
+  CHARACTER(LEN=*),PARAMETER :: myName='init_FM2_Slave'
+  CLASS(FMU2_Slave),INTENT(INOUT) :: self
+  INTEGER(SIK),INTENT(IN) :: id
+  TYPE(ParamType),INTENT(IN) :: pList
+
+  ! Required FMU pList
+  IF(.NOT. pList%has('unzipDirectory')) CALL eFMU_Wrapper%raiseError(modName//'::'//myName//' - No FMU unzipDirectory')
+  CALL pList%get('unzipDirectory', self%unzipDirectory)
+  self%idFMU=id
+  self%FMU_version=2_SIK
+
+  ! Parse the FMU XML model description
+  CALL eFMU_Wrapper%raiseDebug('Opening FMU XML File: '//self%unzipDirectory//'/modelDescription.xml' )
+  CALL self%modelDescription%initFromXML(self%unzipDirectory//'/modelDescription.xml',.TRUE.)
+
+  ! Optional FMU pList
+  IF(pList%has('instanceName')) THEN
+    CALL pList%get('instanceName', self%instanceName)
+  ELSE
+    self%instanceName = "default_fmu_instance"
+  ENDIF
+  IF(pList%has('modelIdentifier')) THEN
+    CALL pList%get('modelIdentifier', self%modelIdentifier)
+  ELSE
+    CALL self%modelDescription%get('modelIdentifier', self%modelIdentifier)
+  ENDIF
+  IF(pList%has('guid')) THEN
+    CALL pList%get('guid', self%guid)
+  ELSE
+    CALL self%modelDescription%get('guid', self%guid)
+  ENDIF
+
+  ! Initilize the FMU
+  self%fmu_c_ptr = InitilizeFMU2_Slave(self%idFMU, &
+    CHAR(self%guid)//c_null_char, &
+    CHAR(self%modelIdentifier)//c_null_char, &
+    CHAR(self%unzipDirectory)//c_null_char, &
+    CHAR(self%instanceName)//c_null_char)
+
+  self%isInit=.TRUE.
+
+ENDSUBROUTINE
+!
+!-------------------------------------------------------------------------------
+!> @brief Set the FMU internal ODE integrator settings
+!>
+!> @param self the FMU2_Slave to act on
+!> @param toleranceDefined flag to override internal FMU ODE tolerance settings
+!> @param tolerance tolerance of FMU ODE integrator
+!> @param startTime start time of FMU ODE integration
+!> @param stopTimeDefined flag to determine if ODE integration stops at stopTime
+!> @param stopTime  end time of ODE integration
+!>
+SUBROUTINE setupExperiment_FMU2_Slave(self, toleranceDefined, tolerance, startTime, stopTimeDefined, stopTime)
+  CLASS(FMU2_Slave),INTENT(INOUT) :: self
```

Comment:
The reason this interface looks like this is because this method signature conforms the the FMI standard interface method: https://github.com/modelica/fmi-standard/blob/1324a73a09bffa488ed40402e69bf2480fb39c0f/headers/fmi2FunctionTypes.h#L179